types/underscore/underscore-tests.ts

```diff
@@ -1005,102 +1006,102 @@ undefinedIdentityIterateeResult; // $ExpectType StringRecord
+    // shallow property iteratee with a non-nullable single type - lists - collect
     _.collect(stringRecordList, stringRecordProperty); // $ExpectType string[]
     _(stringRecordList).collect(stringRecordProperty); // $ExpectType string[]
     extractChainTypes(_.chain(stringRecordList).collect(stringRecordProperty)); // $ExpectType ChainType<string[], string>
 
-    // property name iteratee with a non-nullable single type - dictionaries - map
+    // shallow property iteratee with a non-nullable single type - dictionaries - map
     _.map(stringRecordDictionary, stringRecordProperty); // $ExpectType string[]
     _(stringRecordDictionary).map(stringRecordProperty); // $ExpectType string[]
     extractChainTypes(_.chain(stringRecordDictionary).map(stringRecordProperty)); // $ExpectType ChainType<string[], string>
 
-    // property name iteratee with a non-nullable single type - dictionaries - collect
+    // shallow property iteratee with a non-nullable single type - dictionaries - collect
     _.collect(stringRecordDictionary, stringRecordProperty); // $ExpectType string[]
     _(stringRecordDictionary).collect(stringRecordProperty); // $ExpectType string[]
     extractChainTypes(_.chain(stringRecordDictionary).collect(stringRecordProperty)); // $ExpectType ChainType<string[], string>
 
-    // property name iteratee with other types - lists - map
+    // shallow property iteratee with other types - lists - map
     _.map(stringRecordOrUndefinedList, stringRecordProperty); // $ExpectType any[]
     _.map(intersectingPropertiesList, stringRecordProperty); // $ExpectType (string | boolean)[]
     _.map(nonIntersectingPropertiesList, stringRecordProperty); // $ExpectType any[]
 
-    // property name iteratee with other types - lists - collect
+    // shallow property iteratee with other types - lists - collect
     _.collect(stringRecordOrUndefinedList, stringRecordProperty); // $ExpectType any[]
     _.collect(intersectingPropertiesList, stringRecordProperty); // $ExpectType (string | boolean)[]
     _.collect(nonIntersectingPropertiesList, stringRecordProperty); // $ExpectType any[]
 
-    // property name iteratee - any (see #33479) - map
+    // shallow property iteratee - any (see #33479) - map
     _.map(anyValue, stringRecordProperty); // $ExpectType any[]
     _(anyValue).map(stringRecordProperty); // $ExpectType any[]
     extractChainTypes(_.chain(anyValue).map(stringRecordProperty)); // $ExpectType ChainType<any[], any>
 
-    // property name iteratee - any (see #33479) - collect
+    // shallow property iteratee - any (see #33479) - collect
     _.collect(anyValue, stringRecordProperty); // $ExpectType any[]
     _(anyValue).collect(stringRecordProperty); // $ExpectType any[]
     extractChainTypes(_.chain(anyValue).collect(stringRecordProperty)); // $ExpectType ChainType<any[], any>
 
-    // property path iteratee - lists - map
+    // deep property iteratee - lists - map
```

Comment:
Hold on, I don't think "shallow property"/"deep property" is any more representative of the Underscore documentation than "property name"/"property path". The term "path" does appear in the documentation, as does "key" and all the other terms here.

Not that I object to these terms per se, but it just seems a bit arbitrary.

Reply:
I'm fine with putting in whatever you'd prefer here. Would it be better to use "property key" and "property path"?

Reply:
I like those suggestions. I don't think my preference matters much, though. These are your tests so you can name them however you please (within reasonable limits, but all of these options are reasonable). I'm just a bit confused about the rationale for the change, that's all.

Reply:
The comment I wanted to change most here was "partial object" since "matcher" more accurately describes what the partial object is for.

I got the idea for shallow vs. deep from the last line of the below documentation and thought it might be better than `path` because while that is the parameter name for `_.property` it takes both forms of path.
![image](https://user-images.githubusercontent.com/8692126/89934597-aca8b800-dbc5-11ea-80b0-80caf5172733.png)

Reply:
Just FYI @jgonggrijp it looks like DT is probably set up so that any comment from you retracts your approval, so if you're ok with the rationale above I'd be super grateful if you would please also hit approve again in addition to (and ideally after) making any further replies here 😄

Reply:
Gosh that's restrictive. I'll make sure to hit "approve" again from now on.

Reply:
No worries, just wanted to let you know in advance that that was necessary so you'd only have to make one more trip back here if you were ok with the latest comments 😄 

Thanks for hitting the go button!